### PR TITLE
style-guide: refresh and add shell-specific pages

### PR DESCRIPTION
**Pull request by EmilySeville7cfg**
- [ ] The page (if new), does not already exist in the repository.
- [ ] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [ ] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [ ] The page description includes a link to documentation or a homepage (if applicable).

**Version of the command being documented (if known):**

### CODE REVIEW

File: contributing-guides/style-guide.md [hunk 1]
```diff
@@ -94,6 +98,59 @@ Use backticks on the following:
...
+
+- Define and export environmental variable that persist across shell restarts (builtin):
+
+`set --universal --export {{variable_name}} {{variable_value}}`
+
+- Print the version:
+
+`fish --version`
+```
+
```
Comment (marchersimon) [inline contributing-guides/style-guide.md:153]: Thanks for the effort, but I think that's really too much for a general style guide. If it becomes too long it's hard to find the thing you're searching for.

File: contributing-guides/style-guide.md [hunk 2]
```diff
@@ -66,14 +66,18 @@ Keep the following guidelines in mind when choosing tokens:
...
 - If a particular extension is expected for the file, append it.
   For example, `unrar x {{compressed.rar}}`.
 - In case a generic extension is needed, use `{{.ext}}`, but **only** if an extension is required.
   For instance, in `find.md`'s example "Find files by extension" (`find {{root_path}} -name '{{*.ext}}'`)
   using `{{*.ext}}` explains the command without being unnecessarily specific;
   while in `wc -l {{file}}` using `{{file}}` (without extension) is sufficient.
+  
+### Options
+
+- Prefer long options instead of short ones whenever is possible.
```
Comment (marchersimon) [inline contributing-guides/style-guide.md:80]: I think they're mostly referred to as "flags".
```suggestion
### Flags

- Prefer long flags instead of short ones whenever it is possible. E.g. use `--command`, instead of `-c`. This makes it easier for the user to remember and understand them.
```

Comment (EmilySeville7cfg) [reply contributing-guides/style-guide.md:80]: As I know flags are options without value. That's why I used `options` term.

Comment (CleanMachine1) [reply contributing-guides/style-guide.md:80]: Options is definitely not a good fit, if I was new, I would interpret that as being the token syntax at a glance

File: contributing-guides/style-guide.md [hunk 3]
```diff
@@ -94,6 +98,59 @@ Use backticks on the following:
...
+
+- Start an interactive shell session:
+- Start an interactive shell session without loading startup configs:
+- Execute a command:
+- Execute a script:
+- Check a script for syntax errors:
+- _any other examples_
+- Print the version:
+
+If you want to include shell builtin sample please mark it with `(builtin)` word. For example:
```
Comment (CleanMachine1) [inline contributing-guides/style-guide.md:113]: "What in the world is a shell builtin sample" Me + most others (im guessing)

Even from reading the "For example:" I still have no idea what you are referring to.

File: contributing-guides/style-guide.md (hunk 2 above)
Comment (CleanMachine1) [reply contributing-guides/style-guide.md:80]: I'd reword this to:

```suggestion
### Flags

- Always use long flags when applicable, for example instead of `bash -c`, use `bash --command`, so the reader can understand the meaning of a flag.
```

**Comment by EmilySeville7cfg** [comment]
I am going resolve all suggestions tomorrow. :)

File: contributing-guides/style-guide.md (hunk 3 above)
Comment (EmilySeville7cfg) [reply contributing-guides/style-guide.md]: Now this sentence sounds like:

> If you want to include shell builtin sample please place `(builtin)` word at the end of example description such as:

File: contributing-guides/style-guide.md [hunk 4]
```diff
@@ -66,16 +67,21 @@ Keep the following guidelines in mind when choosing tokens:
...
 - If a particular extension is expected for the file, append it.
   For example, `unrar x {{compressed.rar}}`.
 - In case a generic extension is needed, use `{{.ext}}`, but **only** if an extension is required.
   For instance, in `find.md`'s example "Find files by extension" (`find {{root_path}} -name '{{*.ext}}'`)
   using `{{*.ext}}` explains the command without being unnecessarily specific;
   while in `wc -l {{file}}` using `{{file}}` (without extension) is sufficient.
+  
+### Flags
+
+- Always use long flags when applicable, for example instead of `fish -c`, use `fish --command`, so the reader can understand the meaning of a flag.
```
Comment (sbrl) [inline contributing-guides/style-guide.md:81]: This is a convention yes, but it doesn't always hold true. It's actually a case-by case basis. For example, on pages like `ls` or `git commit` which are extremely frequently used we do not actually recommend `--long-arguments`.

File: contributing-guides/style-guide.md [hunk 5]
```diff
@@ -94,6 +100,28 @@ Use backticks on the following:
 - Extensions, ex. `.dll`.
 - Commands, ex. `ls`.
 
+### Shell-specific pages
```
Comment (sbrl) [inline contributing-guides/style-guide.md:103]: Hrm, I feel like this is overly-specific and isn't a great fit for the style guide. Most pages do not document a shell, so aren't affected by this.

**Comment by EmilySeville7cfg** [comment]
Please reopen pr

File: contributing-guides/style-guide.md [hunk 6]
```diff
@@ -94,6 +100,28 @@ Use backticks on the following:
...
+
+If a page describes some shell (Bash for instance) then when it's possible try follow this structure:
```
Comment (pixelcmtd) [inline contributing-guides/style-guide.md:105]:
```suggestion
If a page describes a shell, when it's possible try to follow this structure:
```

**Review by pixelcmtd — COMMENTED** [review]
I'd still love to drastically shorten the new section, might try that later.

File: contributing-guides/style-guide.md (hunk 5 above)
Comment (git-em) [reply contributing-guides/style-guide.md:105]: Maybe this part could get its own file. There is not that much shells, but there's a lot of builtins.
Even though it's specific, might be good to have this things standardized.

File: contributing-guides/style-guide.md [hunk 7]
```diff
@@ -66,16 +67,23 @@ Keep the following guidelines in mind when choosing tokens:
...
 - If a particular extension is expected for the file, append it.
   For example, `unrar x {{compressed.rar}}`.
 - In case a generic extension is needed, use `{{.ext}}`, but **only** if an extension is required.
   For instance, in `find.md`'s example "Find files by extension" (`find {{root_path}} -name '{{*.ext}}'`)
   using `{{*.ext}}` explains the command without being unnecessarily specific;
   while in `wc -l {{file}}` using `{{file}}` (without extension) is sufficient.
+  
+### Flags
+
+- Always use long flags when available for pages in `linux/` directory. For example instead of `fish -c`, use `fish --command`, so the reader can understand the meaning of a flag.
```
Comment (git-em) [inline contributing-guides/style-guide.md:81]:
```suggestion
- Use long flags when available for pages in `linux/` directory. For example instead of `fish -c`, use `fish --command`, so the meaning of the flag can be understood.
```

File: contributing-guides/style-guide.md (hunk 5 above)
Comment (EmilySeville7cfg) [reply contributing-guides/style-guide.md]: Please view [this](https://github.com/Console-Utils/tldr/blob/shell-guide/contributing-guides/tool-oriented-guides/shells.md) file.

**Comment by EmilySeville7cfg** [comment]
⚠️ This PR is going to be splitted into several PRs to be faster accepted.

File: contributing-guides/style-guide.md [hunk 8]
```diff
@@ -6,27 +6,27 @@ This page lists specific formatting instructions for `tldr` pages.
...
+# <command-name>
 
 > Short, snappy description.
 > Preferably one line; two are acceptable if necessary.
 > More information: <https://example.com>.
 
 - Example description:
 
-`command -opt1 -opt2 -arg1 {{arg_value}}`
+`<command-name> -opt1 -opt2 -arg1 {{arg_value}}`
```
Comment (sbrl) [inline contributing-guides/style-guide.md:18]: This is probably best left as-is.

File: contributing-guides/style-guide.md [hunk 9]
```diff
@@ -6,27 +6,27 @@ This page lists specific formatting instructions for `tldr` pages.
...
 
 - Example description:
 
-`command -opt1 -opt2`
+`<command-name> -opt1 -opt2`
```
Comment (sbrl) [inline contributing-guides/style-guide.md:22]: Same here.

File: contributing-guides/tool-oriented-guides/package-manager.md [hunk 10]
```diff
@@ -0,0 +1,19 @@
+## Package manager page format
+
+If the page describes the package manager, when it's possible try to follow this structure:
+
+```
+- Install the specified packages:
+
+`<command-name>`
```
Comment (sbrl) [inline contributing-guides/tool-oriented-guides/package-manager.md:8]: This `<>` syntax is not a good idea. It confuses things.

**Review by sbrl — COMMENTED** [review]
Some comments, but this is an *enormous* PR
